server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -1108,4 +1116,10 @@ public static boolean assertCurrentMethodIsNotCalledRecursively() {
         }
         return true;
     }
+
+    public static boolean isSystemThreadPool() {
```

Comment:
See my other comment on this. At the very least this code should live in the single place that it's used in IMO to avoid it's use spreading when it's performance is questionable :)